sw/device/lib/testing/test_status.h

```diff
@@ -0,0 +1,63 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_TESTING_TEST_STATUS_H_
+#define OPENTITAN_SW_DEVICE_LIB_TESTING_TEST_STATUS_H_
+
+#include <stdnoreturn.h>
+
+/**
+ * Indicates the status of the software running on the CPU, from a testing
+ * perspective.
+ *
+ * It is useful to track the current status of the test SW executing on the CPU
+ * as a signal the helps aid in debug.
+ *
+ * The values set to these literals is arbitrary and do not possess any special
+ * meaning (other than being hex 'words'). They are chosen to be 16-bit only.
+ * The upper 16-bits are reserved for future use.
+ *
+ * At minimum, it is mandatory to set the status of the software to explicitly
+ * indicate whether the software test passed or failed, with
+ * `kTestStatusPassed` and `kTestStatusFailed` values. As such these are
+ * terimnal states - SW **must not** execute any code after reaching that
+ * status.
```

Comment:
This feels like documentation that might want to go in `set_test_status`.

Also, to be consistent with other documentation, I suggest phrasing the final sentence as "These are terminal states; any code appearing after they are set is unreachable."

In particular, you violate this in the implementation of `test_passed()` and `test_failed()`, by calling `abort()`... personally, I think that `set_test_status` should just call `abort()` itself if called with one of these terminal states.